pkg/systemd/services/services.jsx

```diff
@@ -422,7 +425,19 @@ class ServicesPage extends React.Component {
                                             return Promise.all(result.map(path => this.getUnitByPath(path)));
                                         })
                                         .finally(() => {
-                                            this.setState({ loadingUnits: false });
+                                            // Remove units from state that are not listed from the API in this iteration
+                                            const unit_by_path = Object.assign({}, this.state.unit_by_path);
+                                            let hasExtraEntries = false;
+                                            for (const unitPath in this.state.unit_by_path) {
+                                                if (!this.seenPaths.has(unitPath)) {
+                                                    hasExtraEntries = true;
+                                                    delete unit_by_path[unitPath];
+                                                }
+                                            }
+                                            if (hasExtraEntries)
+                                                this.setState({ unit_by_path });
+
+                                            this.setState({ loadingUnits: false, isFullyLoaded: true });
```

Comment:
Could these two setState calls be combined into one? Like, assign the new state to a variable, and extend it `if hasExtraEntries`.